src/notification/NotificationController.ts

```diff
@@ -0,0 +1,111 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+
+interface viewedNotification {
+  [id: number]: boolean;
+}
+
+interface Notification {
+  id: number;
+  title: string;
```

Comment:
Oh I see - these strings are the _keys_ for the messages, not the messages themselves. Well that's good. Still seems completely unnecessary to store them here though (or anywhere really), since they're all the same (e.g. `'notifications#${id}Title'`)

Reply:
I see that this is still typed to use numeric indexes. But on second thought, this seems fine. Technically they are coerced to strings, but we're allowed to require the use of number indexes regardless.